hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -140,14 +142,22 @@ public static Schema readAvroSchema(Configuration configuration, Path parquetFil
    * Read out the bloom filter from the parquet file meta data.
    */
   public static BloomFilter readBloomFilterFromParquetMetadata(Configuration configuration, Path parquetFilePath) {
-    String footerVal =
-        readParquetFooter(configuration, parquetFilePath, HoodieAvroWriteSupport.HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY)
-            .get(0);
+    List<String> footerVals =
+        readParquetFooter(configuration, false, parquetFilePath,
+            HoodieAvroWriteSupport.HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY,
+            HoodieAvroWriteSupport.OLD_HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY);
+    String footerVal;
+    if (null != footerVals.get(0)) {
+      footerVal = footerVals.get(0);
+    } else {
+      // We use old style key "com.uber.hoodie.bloomfilter"
+      footerVal = footerVals.get(1);
```

Comment:
since required is false here.. if we dont find 0 and 1, then it will NPE? since footerVals have two null entries.. Handling this via nulls still feels risky to me.